templates/fine-tune-llm_v2/end-to-end-examples/fine-tune-preference/README.ipynb

```diff
@@ -0,0 +1,1477 @@
+      "six tennis courts, a football pitch and a playground, to developers to fund\n",
+      "refurbishments, after a £40million revamp under the Building Schools for the\n",
+      "Future programme was cancelled. Alumni of Ark Putney, which used to be Elliott\n",
+      "School, include actor Pierce Brosnan, and 1960s England bowler Geoff Arnold.\n",
+      "Former Welsh secretary Peter Hain sent his children to the school, which was the\n",
+      "scene of the Christmas play in the film Love, Actually. David Cameron has said\n",
+      "his children will attend state school, but George Osborne has been criticised\n",
+      "for sending his to the fee-paying preparatory school in Kensington that he\n",
+      "attended.\n",
+      "\n",
+      "CHOSEN RESPONSE:\n",
+      "Nick Clegg and his wife Miriam are considering sending their 11-year-old son\n",
+      "Antonio to a private school, despite their previous claims that they would use\n",
+      "the state system. Clegg has not visited Ark Putney Academy, a nearby state\n",
+      "school, but its headmaster, Mark Phillips, claims that the school could provide\n",
+      "an 'exceptional' education for any child and that there is no need to pay fees\n",
+      "for schooling. The deadline for applying to Ark Putney for 2013-14 was last\n",
+      "October and the school is part of the Ark academy chain, set up in 2004, whose\n",
+      "chairman is Paul Marshall, one of the Liberal Democrats' biggest donors.\n",
+      "\n",
+      "REJECTED RESPONSE:\n",
+      "Deputy Prime Minister Nick Clegg has defended the possibility of sending his\n",
+      "children to private schools, stating that their education should not be used as\n",
+      "a political football. He and his wife Miriam have not visited their local state\n",
+      "school, but the headmaster of Ark Putney academy in London, where they were\n",
+      "invited to attend, said the school could provide an \"exceptional\" education for\n",
+      "any child and that parents should come to see it firsthand before making a\n",
+      "decision.\n",
+      "\n"
+     ]
+    }
+   ],
+   "source": [
+    "for row in example_rows:\n",
+    "    print_wrapped(\"PROMPT\", row[\"chosen\"][0][\"content\"])\n",
+    "    print_wrapped(\"CHOSEN RESPONSE\", row[\"chosen\"][1][\"content\"])\n",
+    "    print_wrapped(\"REJECTED RESPONSE\", row[\"rejected\"][1][\"content\"])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Step 2: Fine-tuning\n",
+    "\n",
+    "Now that we have the pre-processed dataset, we are ready to fine-tune `Mistral-7B-Instruct-v0.1` using DPO. On Anyscale, we've created an easy-to-use interface to do preference-tuning using `DPO`. We leverage Ray to overlap reference model log-probability calculation with model training to improve GPU utilization. Most implementations compute log probabilities synchronously with model training,\n",
```

Comment:
In here we should make sure they know to read the main LLMForge readme. cross link it.